src/service/node/balancing/load.cpp

```diff
@@ -168,3 +185,47 @@ load_balancer_t::balance() {
         overseer->spawn(pool);
     }
 }
+
+void
+load_balancer_t::on_idle_timer(const std::error_code& ec) {
+    COCAINE_LOG_TRACE(overseer->logger(), "idle timer triggered");
+
+    if (ec == asio::error::operation_aborted) {
+        COCAINE_LOG_TRACE(overseer->logger(), "idle timer, status: aborted");
+        return;
+    }
+
+    COCAINE_LOG_TRACE(overseer->logger(), "checking for idle slaves");
+
+    // TODO: Again, not ideal way to iterate over string->any map. Consider using types.
+    const auto info = overseer->info();
+    const auto pool = info["pool"].as_object();
+    const auto slaves = pool["slaves"].as_object();
+
+    for (const auto& kv : slaves) {
+        const auto& id = kv.first;
+        const auto& slave = kv.second.as_object();
+
+        const auto& state = slave["state"].as_string();
+        COCAINE_LOG_TRACE(overseer->logger(), "state: %s", state);
+
+        if (state == "active") {
+            const auto& load = slave["load"].as_uint();
+            const auto& last_channel_activity = slave["last_channel_activity"].as_int();
+
+            const auto profile = overseer->profile();
```

Comment:
Profiles are planned to be individual for every slave, but not now (it's just not implemented). Instead of this I'd best to take every slave's profile using `overseer->profile(id)` or something like this.